packages/app-desktop/gui/NoteEditor/utils/useFormNote.ts

```diff
@@ -189,13 +189,24 @@ export default function useFormNote(dependencies: HookDependencies) {
 
 			const focusSettingName = noteIsTodo ? 'newTodoFocus' : 'newNoteFocus';
 
-			requestAnimationFrame(() => {
-				if (Setting.value(focusSettingName) === 'title') {
-					if (titleInputRef.current) titleInputRef.current.focus();
-				} else {
-					if (editorRef.current) editorRef.current.execCommand({ name: 'editor.focus' });
+			function autoFocusLoop() {
+				if (!editorRef.current) {
+					// Wait for editorRef to load properly, try again if not loaded yet.
+					// This is necessary to avoid a race condition where auto focus
+					// function is called before the editor is rendered
+					if (!cancelled) setTimeout(autoFocusLoop, 100);
+					return;
 				}
-			});
+
+				requestAnimationFrame(() => {
```

Comment:
One last detail - `requestAnimationFrame` is going to call the callback on the next loop, but by that time the effect might have been unmounted. So I think it means you should check `cancelled` too at the top of this function and exit if necessary.

Reply:
The problem I noticed is that if I add the cancelled check, the auto focus code does not run at all. I previously added more checks to the top of `autoFocusLoop` and `requestAnimationFrame` as well as both of them and they incorrectly do not run the focus functions. Stepping through the debugger, I find out that when `editorRef.current` is undefined, it runs `autoFocusLoop` again and when it does once again, `cancelled` is set to `true`. In my general testing, I have not seen a case where this will be an issue but please let me know of other edge cases where this can be a problem, if any.

Reply:
> I find out that when editorRef.current is undefined, it runs autoFocusLoop again and when it does once again, cancelled is set to true

Yes, but this expected. If it's undefined the effect should be cancelled and you can't run the auto-focus since there's nothing to run it on.

It seems it's currently working but we don't really know why which is not a good sign, because as soon as something is changed somewhere else it's likely to break.

But when I look at it, the whole thing is quite convoluted anyway. Why not create `setInterval` and run your code in there. Just wait till the refs are defined and then set the focus. Once that's done clear the interval. Maybe the reason it fails is that you check for editorRef, but not for titleInputRef.

Reply:
I revised my approach to the issue and instead of waiting for a loop, since it seems to just be a timing issue due to the race condition, I just used a delay instead of an interval or a timeout recursive loop. I switched to this because mainly it really doesn't need to check multiple times and just needs an appropriate delay to fix the timing. I found that 50ms produces the most reliable results without sacrificing too much speed, below that I've found out it doesn't seem to work all the time. I'm not entirely sure if the focused flag is entirely necessary as well but given what you mentioned I decided I'd add it in there to be safe. Let me know if you think otherwise, then it should be really simple to remove.